api/v1alpha1/onload_types.go

```diff
@@ -4,25 +4,76 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// Currently unimplemented
+type OnloadBuildSpec struct {
+}
+
+// Currently unimplemented
+type DevicePluginBuildSpec struct {
+}
+
 // OnloadSpec defines the desired state of Onload
 type OnloadSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// KernelImage
+	// An image that contains the out-of-tree kernel modules used by onload,
+	// including the sfc driver.
+	KernelImage string `json:"kernelImage"`
+
+	// UserImage
+	// An image that contains the built userland objects, used by the cplane
+	// and deviceplugin DaemonSets.
+	UserImage string `json:"userImage"`
+
+	// DevicePluginImage
+	DevicePluginImage string `json:"devicePluginImage"`
+
+	// +optional
+	// Optional version string to associate with this onload CR.
+	Version string `json:"version,omitempty"`
 
-	// Foo is an example field of Onload. Edit onload_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// +optional
+	// ImagePullPolicy is the policy used when pulling images.
+	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
+	ImagePullPolicy v1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// +optional
+	// Build contains build options.
+	// If non-nil the operator will attempt to build the required images using
+	// what is defined in the build section, and push the images to the
+	// specified locations.
+	Build *OnloadBuildSpec `json:"build,omitempty"`
+
+	// +optional
+	// Specific instructions for building the device plugin image.
+	DevicePluginBuild *DevicePluginBuildSpec `json:"devicePluginBuild,omitempty"`
```

Comment:
Do we want to move it to the (new) `DevicePluginSpec` along with `DevicePluginImage`?